client/keys/migrate_test.go

```diff
@@ -2,41 +2,97 @@ package keys
 
 import (
 	"context"
-	"fmt"
+     "fmt"
 	"testing"
 
+	design99keyring "github.com/99designs/keyring"
+	"github.com/stretchr/testify/suite"
+
 	"github.com/cosmos/cosmos-sdk/client"
+    "github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/multisig"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/simapp"
+   "github.com/cosmos/cosmos-sdk/testutil"
+	"github.com/cosmos/cosmos-sdk/codec"
+)
 
-	"github.com/stretchr/testify/require"
+type setter interface {
+	SetItem(item design99keyring.Item) error
+}
 
-	"github.com/otiai10/copy"
-	"github.com/stretchr/testify/assert"
+type MigrateTestSuite struct {
+	suite.Suite
 
-	"github.com/cosmos/cosmos-sdk/client/flags"
-	"github.com/cosmos/cosmos-sdk/crypto/keyring"
-	"github.com/cosmos/cosmos-sdk/testutil"
-)
+	dir  string
+	n1   string
+	cdc codec.Codec
+	priv cryptotypes.PrivKey
+	pub  cryptotypes.PubKey
 
-func Test_runMigrateCmd(t *testing.T) {
-	kbHome := t.TempDir()
-	clientCtx := client.Context{}.WithKeyringDir(kbHome)
-	ctx := context.WithValue(context.Background(), client.ClientContextKey, &clientCtx)
+}
+
+func (s *MigrateTestSuite) SetupSuite() {
+	s.dir = s.T().TempDir()
+	s.cdc = simapp.MakeTestEncodingConfig().Codec
+	s.n1 = "cosmos"
+	s.priv = cryptotypes.PrivKey(secp256k1.GenPrivKey())
+	s.pub = s.priv.PubKey()
+}
 
-	require.NoError(t, copy.Copy("testdata", kbHome))
 
-	cmd := MigrateCommand()
+func (s *MigrateTestSuite) Test_runListAndShowCmd() {
```

Comment:
Should I add more CLI tests here?

Reply:
yes, the  name is good.

We should have a unit test which works directly on the structures without going through the command.

Reply:
You mean [these tests](https://github.com/cosmos/cosmos-sdk/blob/cyberbono3/keyring-migration/crypto/keyring/migration_test.go)?

Reply:
I added Migrate CLI command [tests](https://github.com/cosmos/cosmos-sdk/blob/cyberbono3/keyring-migration/client/keys/migrate_test.go#L93-L150) as well. Does it sound good or you feel I have to improve testing coverage?